sdk/options.go

```diff
@@ -0,0 +1,101 @@
+package tfsdklog
+
+import (
+	"io"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+// Option defines a modification to the configuration for a logger.
+type Option func(loggerOpts) loggerOpts
```

Comment:
In order to share these between the two packages, we'd need to export `loggerOpts` and all the properties on it, and I think I'd prefer not to do that. I think the duplication here isn't bad, and a little copying may be preferable to a little dependency. Especially if we want to introduce options that only apply to one of these logging contexts.

Reply:
Another option would be to move that to an exported Go type in an `internal/` package. Not the biggest deal though, in my opinion. My initial reaction when reviewing this was "what is different between the two `Option` implementations" but sounds like there is a future use case.

Reply:
Yeah, I think that's a potential future direction. I think my justification here is that though they look the same now, the audiences are different, and so the options may differ in the future. For example, setting the output isn't something provider devs should really worry about, but it may make sense for SDK devs.

But then again, both uses of options are really meant so SDK devs, so I don't know.

I think, given that it's a private, internal-only detail, it's probably fine for the moment, because refactoring will have basically no user-visible changes, so we're always able to refactor it into an internal package later.